### PR TITLE
docs: make setting `gas` optional and require evmVersion

### PR DESCRIPTION
**Pull request by kabaluyot**

### CODE REVIEW

File: versioned_docs/version-23.10.2/private-networks/how-to/configure/free-gas.md [hunk 1]
```diff
@@ -136,8 +136,23 @@ Update the `hardhat.config.js` file:
    gasPrice: 0;
    ```
 
-1. Set the gas limit for a transaction (that is, contract creation) to be the block gas limit - 1.
+2. Set the gas limit for a transaction (that is, contract creation) to be the block gas limit - 1.
```
Comment (joaniefromtheblock) [inline versioned_docs/version-23.10.2/private-networks/how-to/configure/free-gas.md:139]:
```suggestion
1. Set the gas limit for a transaction (that is, contract creation) to be the block gas limit - 1.
```

Comment (joaniefromtheblock) [reply versioned_docs/version-23.10.2/private-networks/how-to/configure/free-gas.md:139]: Markdown should auto increment the numbering

File: versioned_docs/version-23.10.2/private-networks/how-to/configure/free-gas.md [hunk 2]
```diff
@@ -136,8 +136,23 @@ Update the `hardhat.config.js` file:
...
 
    ```js
    gas: "0x1ffffffffffffe";
    ```
+   
+> In most deployments, setting the `gasPrice` to 0 is enough (if contract is not that big).
```
Comment (joaniefromtheblock) [inline versioned_docs/version-23.10.2/private-networks/how-to/configure/free-gas.md:145]:
```suggestion
      > For most deployments, setting `gasPrice` to `0` should cover transaction costs, especially if the contract is not large.
```

File: versioned_docs/version-23.10.2/private-networks/how-to/configure/free-gas.md [hunk 3]
```diff
@@ -136,8 +136,23 @@ Update the `hardhat.config.js` file:
...
+
+
+3. Specify the `evmVersion` when using latest Solidity versions:
```
Comment (joaniefromtheblock) [inline versioned_docs/version-23.10.2/private-networks/how-to/configure/free-gas.md:148]:
```suggestion
3. Specify `evmVersion` when using the latest Solidity version:
```

File: versioned_docs/version-23.10.2/private-networks/how-to/configure/free-gas.md [hunk 4]
```diff
@@ -136,8 +136,23 @@ Update the `hardhat.config.js` file:
...
+
+   ```js
+   solidity: {
+      version: "0.8.20",
+      settings: {
+        evmVersion: "london", // required for besu
```
Comment (joaniefromtheblock) [inline versioned_docs/version-23.10.2/private-networks/how-to/configure/free-gas.md:154]:
```suggestion
        evmVersion: "london", // required for Besu
```

**Comment by kabaluyot** [comment]
Looks like I edited a versioned docs .. Creating new PR and addressing revisions there: https://github.com/hyperledger/besu-docs/pull/1498

**Pull request closed without merging**